pulpcore/pulpcore/app/viewsets/content.py

```diff
@@ -79,3 +81,24 @@ class ArtifactViewSet(CreateDestroyReadNamedModelViewSet):
     queryset = Artifact.objects.all()
     serializer_class = ArtifactSerializer
     filter_class = ArtifactFilter
+
+    def destroy(self, request, pk):
+        """
+        Remove :class:`~pulpcore.app.models.Artifact` only if it doesn't belong to any
+        `~pulpcore.app.models.Content`
+
+        Args:
+            request (:class:`rest_framework.request.Request`): request containing JSON payload
+                with information about the artifact being deleted.
+            pk (:class:`uuid.UUID`): Unique id of the artifact being deleted.
+
+        Returns:
+            :class:`rest_framework.response.Response` with a 204 status code in case of
+                successful deletion, 409 status code if deletion is not allowed.
+        """
+        try:
+            return super(ArtifactViewSet, self).destroy(request, pk)
+        except models.ProtectedError:
```

Comment:
@jortel , does it make sense to return the error message that `ProtectedError` creates:  
`("Cannot delete some instances of model 'Artifact' because they are referenced through a protected foreign key: 'ContentArtifact.artifact'", <QuerySet [<ContentArtifact: pk=ebf4eea0-7942-4b22-8ad4-f63ba944899d>]>)` instead on the vague one I wrote?
But I wonder if queryset contains many entries... maybe at least log it at debug log level? What do you think?

Reply:
I think catching the `Protected` exception and including a _user facing_ description as you did is better.  However, I would suggest a small change in wording.

_'The Artifact cannot be deleted because it is associated with Content.'_